lib/matplotlib/colorbar.py

```diff
@@ -716,12 +726,28 @@ def set_ticklabels(self, ticklabels, update_ticks=True):
         Tick labels are updated immediately unless *update_ticks* is *False*,
         in which case one should call `.update_ticks` explicitly.
         """
-        if isinstance(self.locator, ticker.FixedLocator):
+        # check if explitic ticks have been supplied
+        if self._ticks is None:
+            cbook._warn_external('To set explicit ticklabels, call colorbar() '
+                                 'with ticks keyword or use set_ticks() '
+                                 'method first.')
+        # check if length of ticks and ticklabels match
+        elif len(self._ticks) != len(self._ticklabels):
+            cbook._warn_external('The ticklabels need to be of the same '
+                                 'length as the ticks.')
+        # check if objects in list have valid type
+        elif not all(type(item) in [str, float, int] for item in ticklabels):
+            cbook._warn_external('ticklabels need to be a list of str')
```

Comment:
It would be good to check these warnings and the next one if you have time for that....

Reply:
Actually, w/o reading the discussion, do we need all these warning checks?  I don't think we usually check for mismatched type, for instance...